package.json

```
@@ -7,6 +7,9 @@
   "module": "dist/qss.m.js",
   "main": "dist/qss.js",
   "types": "qss.d.ts",
+  "exports": {
+    "import": "./dist/qss.mjs"
+  },
```

Note that adding "exports" is a breaking change, and to be maximally backwards compatible you need a string fallback, as well as the "default" condition specified.

Also, as-is, this will break, so here's how you'd need it to look:
```suggestion
  "exports": {
    ".": [
      {
        "import": "./dist/qss.mjs",
        "require": "dist/qss.js",
        "default": "dist/qss.js"
      },
      "dist/qss.js"
    ],
    "./package.json": "./package.json"
  },
```

---

Whoa, thank you for catching that.

---

Array exports are the least compatible of all types. Please prepare an object only. As mentioned, this is a breaking change by the mere presence of “exports” alone due to wildly varying resolution support across early Node versions, so not going to step into “exports” territory with the buggiest/least-supported 1st foot.

---

@lukeed i'm not sure what you mean, *every single* implementation of the "exports" field supports array fallbacks, and i use it on hundreds of packages with zero reported issues.

---

I have added more commits.
Changes:
* Added a `.cjs` file so that require from a `type: module` can work. I.E. fixes error: "To treat it as a CommonJS script, rename it to use the '.cjs' file extension."
* Now using `.mjs` instead of `.m.js` to be consistent. Currently, there is a `dist/qss.mjs` but I couldn't figure out how it gets generated, so instead, I just renamed the only file I could see, `.m.js` to `.mjs`.
* The exports object is only using features written here: https://nodejs.org/api/packages.html#exports. I couldn't find docs on using arrays, so I don't feel too confident using them.
* The exports object allows direct importing of the full file path, including extension. I think this should make it backwards compatible.
* I tested and found these working:
  * importing from a type: module project
  * requiring from a type: commonjs project
  * direct imports to full path w/ extension

---

Note that without the array, the engines field should be modified to exclude node 13.0-13.2.